server/aws/lambda/aggregate_values.js

```diff
@@ -78,31 +149,59 @@ function pinDate(timestamp) {
 function aggregateEvents(event){
     const aggregates = {}; 
     event.Records.forEach((record) => {
+        try { 
+            if(record.eventName === 'INSERT') {
+                const raw = JSON.parse(record.dynamodb.NewImage.raw.S);
+                raw.payload.forEach((pl) => {
 
-        if(record.eventName === 'INSERT') {
-            const raw = JSON.parse(record.dynamodb.NewImage.raw.S);
-            raw.payload.forEach((pl) => {
-                const date = pinDate(pl.timestamp);
-                const sk = createSK(date, raw.appversion, raw.appos, pl);
+                    // bucket values to reduce possible permutations
+                    const date = pinDate(pl.timestamp);
+                    pl.count = bucketCount(pl.count);
+                    pl.duration = bucketDuration(pl.duration);
 
-                if (sk in aggregates){
-                    aggregates[sk].metricCount = aggregates[sk].metricCount + 1;
+                    // deal with potentially missing data
+                    const osversion = raw.osversion || '';
+                    const manufacturer = raw.manufacturer || '';
+                    const androidreleaseversion = raw.androidreleaseversion || '';
 
-                } else {
+                    const sk = createSK (
+                            date,
+                            raw.appversion,
+                            raw.appos,
+                            osversion,
+                            manufacturer,
+                            androidreleaseversion,
+                            pl
+                        );
 
-                    aggregates[sk] = {
-                        ...pl,
-                        pk: pl.region,
-                        sk: createSK(date, raw.appversion, raw.appos, pl),
-                        date: date,
-                        appos: raw.appos,
-                        appversion: raw.appversion,
-                        metricCount: 1
-                    };
+                    if (sk in aggregates){
 
-                }
+                        aggregates[sk].metricCount = aggregates[sk].metricCount + 1;
 
-            });
+                    } else {
+
+                        aggregates[sk] = {
+                            ...pl,
+                            pk: pl.region,
+                            sk: sk,
+                            date: date,
+                            appos: raw.appos,
+                            appversion: raw.appversion,
+                            osversion: osversion,
+                            metricCount: 1,
+                            manufacturer: manufacturer,
+                            androidreleaseversion: androidreleaseversion
+                        };
+
+                    }
+
+                });
+            }
+
+        } catch(err) {
+            console.err(`issue parsing event: ${err}`);
```

Comment:
I feel like you're trolling me here. :)

console.error() oui?

Reply:
Fixed